src/components/Advertising/utils/createAdvertisingSessionManager.js

```
@@ -0,0 +1,133 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { getNamespacedCookieName, cookieJar } from "../../../utils/index.js";
+import createLoggingCookieJar from "../../../utils/createLoggingCookieJar.js";
+import {
+  ADVERTISING_COOKIE_KEY,
+  DEFAULT_COOKIE_EXPIRATION_MINUTES,
+  DEFAULT_THROTTLE_MINUTES,
+} from "../constants/index.js";
+
+export default ({ orgId, logger }) => {
+  const loggingCookieJar = createLoggingCookieJar({ logger, cookieJar });
+
+  const getCookieName = (key, useNamespace = true) =>
+    useNamespace ? getNamespacedCookieName(orgId, key) : key;
+
+  const getDefaultExpiration = (minutes = DEFAULT_THROTTLE_MINUTES) =>
+    new Date(Date.now() + minutes * 60 * 1000);
+
+  const safeJsonParse = (value) => {
+    try {
+      if (value?.startsWith("%7B") || value?.startsWith("{")) {
+        return JSON.parse(decodeURIComponent(value));
+      }
+    } catch {
+      // Silent catch for invalid JSON - return original value
+    }
+    return value;
+  };
+
+  const safeJsonStringify = (value) =>
+    typeof value === "object" && value !== null
+      ? encodeURIComponent(JSON.stringify(value))
+      : value;
+
+  const readCookie = (key, useNamespace = true) => {
+    try {
+      const name = getCookieName(key, useNamespace);
+      const value = loggingCookieJar.get(name);
+      return value ? safeJsonParse(value) : null;
+    } catch (error) {
```

I don't think these try/catches are necessary around loggingCookieJar.get

---

loggingCookieJar.get is implemented using js-cookie and may throw exceptions in the following scenarios:

- Invalid cookie names
- 
- Browser security restrictions
- 
- Storage quota exceeded
- 
- Third-party cookie restrictions

To handle these edge cases gracefully, a try-catch block is used. If accessing the cookie fails, the error is ignored and null is returned.